diff --git a/node_modules/redux-persist/lib/persistReducer.js b/node_modules/redux-persist/lib/persistReducer.js
index 1116881..e76bf6a 100644
--- a/node_modules/redux-persist/lib/persistReducer.js
+++ b/node_modules/redux-persist/lib/persistReducer.js
@@ -5,7 +5,7 @@ exports.default = persistReducer;
 
 var _constants = require("./constants");
 
-var _autoMergeLevel = _interopRequireDefault(require("./stateReconciler/autoMergeLevel1"));
+var _autoMergeLevel2 = _interopRequireDefault(require("./stateReconciler/autoMergeLevel2"));
 
 var _createPersistoid = _interopRequireDefault(require("./createPersistoid"));
 
@@ -41,7 +41,7 @@ function persistReducer(config, baseReducer) {
 
   var version = config.version !== undefined ? config.version : _constants.DEFAULT_VERSION;
   var debug = config.debug || false;
-  var stateReconciler = config.stateReconciler === undefined ? _autoMergeLevel.default : config.stateReconciler;
+  var stateReconciler = config.stateReconciler === undefined ? _autoMergeLevel2.default : config.stateReconciler;
   var getStoredState = config.getStoredState || _getStoredState.default;
   var timeout = config.timeout !== undefined ? config.timeout : DEFAULT_TIMEOUT;
   var _persistoid = null;
diff --git a/node_modules/redux-persist/types/createMigrate.d.ts b/node_modules/redux-persist/types/createMigrate.d.ts
index 2c1329d..2e4cd57 100644
--- a/node_modules/redux-persist/types/createMigrate.d.ts
+++ b/node_modules/redux-persist/types/createMigrate.d.ts
@@ -11,7 +11,7 @@ declare module "redux-persist/es/createMigrate" {
    * @param config migration configuration
    */
   // tslint:disable-next-line: strict-export-declare-modifiers
-  export default function createMigrate(migrations: MigrationManifest, config?: MigrationConfig): PersistMigrate;
+  export default function createMigrate<T extends {}>(migrations: MigrationManifest<T>, config?: MigrationConfig): PersistMigrate;
 }
 
 declare module "redux-persist/lib/createMigrate" {
diff --git a/node_modules/redux-persist/types/types.d.ts b/node_modules/redux-persist/types/types.d.ts
index b3733bc..f8106cb 100644
--- a/node_modules/redux-persist/types/types.d.ts
+++ b/node_modules/redux-persist/types/types.d.ts
@@ -72,9 +72,9 @@ declare module "redux-persist/es/types" {
     removeItem(key: string): Promise<void>;
   }
 
-  interface MigrationManifest {
-    [key: string]: (state: PersistedState) => PersistedState;
-  }
+  type MigrationFunction<T extends {}, U extends {} = T> = T extends infer V ? U extends infer W ? (state: V) => W | Promise<W> : never : never;
+
+  type MigrationManifest<T extends {}> = Record<string, MigrationFunction<T>>;
 
   /**
    * @desc
